lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/java/AvroClassHierarchy.java

```diff
@@ -0,0 +1,200 @@
+package org.apache.reef.tang.implementation.java;
+
+import org.apache.reef.tang.ClassHierarchy;
+import org.apache.reef.tang.annotations.NamedParameter;
+import org.apache.reef.tang.exceptions.NameResolutionException;
+import org.apache.reef.tang.formats.avro.*;
+import org.apache.reef.tang.implementation.types.*;
+import org.apache.reef.tang.types.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Implementation of ClassHierarchy for Avro
+ */
+public class AvroClassHierarchy implements ClassHierarchy {
```

Comment:
AvroClassHierarchy is used by AvroClassHierarchySerializer which is in different namespace. It cannot be private in this case. However, I don't think we should put AvroClassHierarchySerializer in Format folder. Format is mainly used for Configuration Modules. 
One option is similar to protobuf, create a subfolder avro under implementation and leave both AvroClassHierarchy and AvroClassHierarchySerializer. AvroClassHierarchy  essentially does deserialize work.